addons/website_event/views/event_templates_list.xml

```diff
@@ -349,11 +349,16 @@
                             </small>
                         </div>
                         <!-- Location -->
-                        <t t-set="has_city_or_country" t-value="event.address_id.sudo().city or event.address_id.sudo().country_id"/>
-                        <div t-if="not event.address_id or has_city_or_country" class="d-flex align-items-center">
+                        <div t-if="event.address_id" class="d-flex align-items-center">
+                            <t t-set="location_with_state" t-value="'%s'%(', '.join(([event.address_id.sudo().city] if event.address_id.sudo().city else []) + [i.name for i in [event.address_id.sudo().state_id, event.address_id.sudo().country_id] if i]))"/>
+                            <t t-set="location_without_state" t-value="'%s'%(', '.join(([event.address_id.sudo().city] if event.address_id.sudo().city else []) + ([event.address_id.sudo().country_id.name] if event.address_id.sudo().country_id else [])))"/>
+                            <i t-if="event.address_id.country_id.state_required or location_without_state" class="fa fa-map-marker me-2" title="Location"/>
+                            <small t-if="event.address_id.country_id.state_required" class="o_not_editable fw-bold" itemprop="location" t-out="location_with_state"/>
+                            <small t-else="" class="o_not_editable fw-bold" itemprop="location" t-out="location_without_state"/>
+                        </div>
+                        <div t-else="" class="d-flex align-items-center">
```

Comment:
Hello @adsh-odoo Can't we do this for better readability and case when we only have a street? Didn't check functional cases and this is just an idea If you have a better optimized way for this then go for it.
```suggestion
      <t t-set="has_city" t-value="event.address_id.sudo().city"/>
      <t t-set="has_state" t-value="event.address_id.sudo().state_id and event.address_id.country_id.state_required"/>
       <t t-set="has_country" t-value="event.address_id.sudo().country_id"/>
            <div t-if="has_city or has_state or has_country" class="d-flex align-items-center">
                  <t t-set="location" t-value="[]"/>
                      <t t-if="has_city">
                          <t t-set="location" t-value="[event.address_id.sudo().city]"/>
                       </t>
                        <t t-if="has_state">
                           <t t-set="location" t-value="location + [event.address_id.sudo().state_id.name]"/>
                         </t>
                         <t t-if="has_country">
                            <t t-set="location" t-value="location + [event.address_id.sudo().country_id.name]"/>
                           </t>
                            <i class="fa fa-map-marker me-2" title="Location"/>
                            <small class="o_not_editable fw-bold" itemprop="location" t-out="', '.join(location)"></small>
                        </div>
                        <div t-elif="not event.address_id" class="d-flex align-items-center">
                            <i class="fa fa-map-marker me-2" title="Location"/>
                            <small class="o_not_editable fw-bold" itemprop="location">Online event</small>
                        </div>
```